src/main/java/org/verdictdb/jdbc41/VerdictStreamResultSet.java

```diff
@@ -884,7 +890,12 @@ public boolean next() throws SQLException {
       } else if (queryResults.peek() == null && hasReadAllQueryResults) {
         return false;
       } else {
-        queryResult = queryResults.take();
+        log.debug("Trying to take a queryResult out of blockingDeque");
+        queryResult = queryResults.poll(2, TimeUnit.MINUTES);
```

Comment:
It seems deadlock does appear after I commented those lines. I have added some logic after this when it fails to take a token. It will check `hasReadAllQueryResults` again to see whether it is still in progress. If `hasReadAllQueryResults` is true already, we then return false to indicate no result left.

Reply:
I have run tests for multiple times and the fix seems work.